packages/plugins/typescript/urql/src/visitor.ts

```diff
@@ -0,0 +1,75 @@
+import { ClientSideBaseVisitor, ClientSideBasePluginConfig, getConfigValue } from '@graphql-codegen/visitor-plugin-common';
+import { UrqlRawPluginConfig } from './index';
+import * as autoBind from 'auto-bind';
+import { FragmentDefinitionNode, OperationDefinitionNode, Kind } from 'graphql';
+import { toPascalCase } from '@graphql-codegen/plugin-helpers';
+import { titleCase } from 'change-case';
+
+export interface UrqlPluginConfig extends ClientSideBasePluginConfig {
+  withComponent: boolean;
+  withHooks: boolean;
+  urqlImportFrom: string;
+}
+
+export class UrqlVisitor extends ClientSideBaseVisitor<UrqlRawPluginConfig, UrqlPluginConfig> {
+  constructor(fragments: FragmentDefinitionNode[], rawConfig: UrqlRawPluginConfig) {
+    super(fragments, rawConfig, {
+      withComponent: getConfigValue(rawConfig.withComponent, true),
+      withHooks: getConfigValue(rawConfig.withHooks, false),
+      urqlImportFrom: getConfigValue(rawConfig.urqlImportFrom, null),
+    } as any);
```

Comment:
I think you should be able to remove `as any` now.

Reply:
Done!